packages/components/addon/components/hds/copy/button/index.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export const DEFAULT_SIZE = 'medium';
+export const DEFAULT_COLOR = 'secondary';
+export const SIZES = ['small', 'medium'];
+export const COLORS = ['secondary'];
+
+export default class HdsCopyButtonComponent extends Component {
+  @tracked isSuccess = false;
+  @tracked isError = false;
+
+  /**
+   * @param text
+   * @type {string}
+   * @description The text to be used for the accessible name of the button. If no text value is defined an error will be thrown
+   */
+  get text() {
+    let { text } = this.args;
+
+    assert(
+      '@text for "Hds::Copy::Button" must have a valid value',
+      text !== undefined
+    );
+
+    return text;
+  }
+
+  /**
+   * @param clipboardText
+   * @type {string}
+   * @description The ID of the element containing the text to be copied. If no clipboardText value is defined `isError` will be set to true.
+   */
+  get clipboardText() {
+    let { clipboardText } = this.args;
+    return clipboardText;
+  }
+
+  /**
+   * @param size
+   * @type {string}
+   * @default medium
+   * @description The size of the copy/button; acceptable values are `small` and `medium`
+   */
+  get size() {
+    let { size = DEFAULT_SIZE } = this.args;
+
+    assert(
+      `@size for "Hds::Copy::Button" must be one of the following: ${SIZES.join(
+        ', '
+      )}; received: ${size}`,
+      SIZES.includes(size)
+    );
+
+    return size;
+  }
+
+  /**
+   * @param color
+   * @type {string}
+   * @default secondary
+   * @description Determines the color of copy/button to be used; the only acceptable value is `secondary`
+   */
+  get color() {
+    let { color = DEFAULT_COLOR } = this.args;
+
+    assert(
+      `@color for "Hds::Copy::Button" must be one of the following: ${COLORS.join(
+        ', '
+      )}; received: ${color}`,
+      COLORS.includes(color)
+    );
+
+    return color;
+  }
+
+  /**
+   * @param isIconOnly
+   * @type {boolean}
+   * @default false
+   * @description Indicates if the button will only contain an icon; component will also ensure that accessible text is still applied to the component.
+   */
+  get isIconOnly() {
+    return this.args.isIconOnly ?? false;
+  }
+
+  /**
+   * @param isFullWidth
+   * @type {boolean}
+   * @default false
+   * @description Indicates that a button should take up the full width of the parent container. The default is false.
+   */
+  get isFullWidth() {
+    return this.args.isFullWidth ?? false;
+  }
```

Comment:
A lot of the arguments are duplicated from `Hds::Button`. Was there a strong reason to do that instead of using that component?